src/main/java/io/qmbot/aoc/y2022/Day17.java

```diff
@@ -0,0 +1,249 @@
+package io.qmbot.aoc.y2022;
+
+import io.qmbot.aoc.Puzzle;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class Day17 implements Puzzle {
+    @Override
+    public String part1(String input) {
+        String[][] field = new String[24276][9];
+
+        fullField(field);
+
+        int startDirection = 0;
+        int start = field.length - 5;
+        int indexF = 0;
+
+        for (int i = 0; i < 2022; i++) {
+            for (String[] string : field) {
+                if (Arrays.asList(string).contains("7")) {
+                    start = Arrays.asList(field).indexOf(string) - 4;
+                    break;
+                }
+            }
+
+            startDirection = createFigure(indexF, start).startFalling(field, input, startDirection);
```

Comment:
In my case 'input' was ended by EOL symbol '\n', and this caused an exception in lines 199-200. I suggest to trim input, e.g. use 'input.trim()' instead of 'input' here